### ISSUE TITLE
[NIP23]: Taxation - Treasury - Burn

### ISSUE THREAD

**Issue by IWantMoreDecred** [NeblioTeam/Neblio-Improvement-Proposals#23]
### Proposal Description

[Updated based on the comment i have received.]

Staking is currently:  10% to Node, 10% to cold-staker [which could be anyone. strangers or yourself]
And currently we have no burn.  Also, we have no easy way for others to delegate to nodes. 

Problem?
1. We need to incentivize more people to choose being a node over being a cold-staker [unless you are cold-staking on your OWN NODE. In which case, a "White List" could be also added in for your own stuff.]
2. Some think we should have a Treasury address to help aid in paying for things.
3. Try to battle inflation
4. Make Staking easier so Pooled staking delegations possible.

Solution:  New Staking system should be as follows:  Still the SAME 10% to Node, 10% to cold-staker.....   but with a catch.  
[NEW]
1. TREASURY TAX of 10% added to Cold-Staker's rewards, but ONLY to Cold-Stakers who are NOT WHITE-LISTed are subject to this taxation, because they might be cold-staking on their own node.  Basically, this would TTax anyone who isn't a node to incentivize being a node, but if they still wish to not node, they would have a base earning of 9% instead of 10%, assuming they are piggy-backing off a node that is not charging any fee, as described below in the other items of this proposal.

This will give them incentive to be a node, and when they choose not to, then they are helping Treasury instead.   [Another way to look at it is that the Node, because they are performing a service to us all, are Tax-exempt]
This way, either option helps Neblio in some way in return for the staking rewards.  So, the way it works would be, each and every time a staking reward comes to the Cold-Staker, 10% of that 10% reward is deducted first and sent to a Treasury address, leaving them with 9% yearly in rewards. 

2.  Add feature in NEblio QT and other node wallets to allow Nodes to set up their own custom Fee % that also targets the Cold-Staker's rewards, but only AFTER Treasury Taxes are collected from the cold-staker first.  Of what remains of the rewards, the Node can choose a Fee percent from 0.00%-100.00% of the Cold-staker's rewards to further incentivize being a Node.  Since a lot of Cold-stakers are really just delegators to a pool, and since some nodes have Cold-staking set up on their own node so that none of their wallet is HOT for higher security, then a White-List should also be in place so that the Nodes can White-List their own cold-Self-delegation to make addresses Fee-exempt too. 

3. Nodes should see the address of each Cold-wallet that is delegated to them. WHY? ISPOs are then possible to send token rewards to delegators! -  This stands for "Initial Stake Pool Offering". It is how Sundae Swap & Meld, on the Cardano network, earned Their funding... so i'd like to see a similar system in place.  
Example:  I participated in Sundae Swap ISPO by delegating my ADA to their node, where they set their fee to 100%.  All my ADA rewards went to them in that time and after the ISPO duration ended after a couple of months I then received Sundae tokens.
We could do ISPOs too then if we had Nodes collecting custom fees & know the addresses that are staked to them in order to send them tokens if we have our own project too.  Naturally most nodes would not be running an ISPO but it's good to have the option in case we ever started suddenly having lots of dapps/defi one day.
Lets use this idea too because it works very well in cardano.

4. Create a "Stake" tab in Orion web wallet.  Clicking it will reveal a list of all Nodes & their fees for them to choose from. Plus it could display other data,.. such as downtime/uptime.. description, or whatever would let a cold-staker know a good Node to pool to versus a bad Node to pool to or let them know what might be holding an "ISPO"  You could probably get some ideas from how cardano or how Terra Luna's system works in their Terra Station.  Located here:  https://station.terra.money/
They have a nice setup/layout on Terra. Orion's design could borrow some ideas i think. 

6. And as far as Inflation goes. The same kind of Burn that Ethereum has should be added.. so that as Neblio's chain gains more and more activity, the Burn will work the same and burn more and more.. The more activity, the more the burn. This could be done as a base flat fee or %. but i'm not sure what would work best with Neblio's unique setup but maybe stick exactly to how ETH does it if it's similar enough? Fine with me. I just want to curve this over the long haul into a deflationary currency as the chain becomes active more.

All incoming Stake Rewards are NOT subject to the Burn-tax. All other activity is Burn-taxed and sent to the furnace, in the same way Ethereum does it. Basically "Sends", "NPT1 token creation", "Dapp/smart contract creations & executions" & typical stuff.. Just not Stake Rewards. Nodes are NOT to be punished in any way. 

=========================Conclusion:===========================

This fuels a Treasury while incentivizing people to choose being a Node over cold-staking.  When they still choose to cold-stake, they are at least helping treasury.  Nodes would be Tax-exempt for performing a service & can collect Fee % of the rewards of any delegations pooled to them, enabling also the possibility of ISPO's being done on Neblio.  And Orion is added the functionality of cold-staking and there is a marketplace added for very easily picking a node to pool to.  For nodes that are Self-delegated-Cold-Staking for security reasons, they would have a White-List option added to the wallet for them to use to whitelist any addresses they want TTax/Fee-exempt.

Having a Treasury would not be something devs can fully depend on, but it would give them a modest sum to work with to do certain things. Maybe contests? Bug Bounty?  I don't know what the current amount of cold-stakers are, but lets use an example:

With the Treasury 10% on rewards tax, for every 10 NEBL that comes in to all the Cold-Stakers, they would receive 9 instead of 10, with 1 going to treasury. As an individual cold-staker, you might receive 1NEBL in a single stake reward sometimes as is, but with the TTax you would see .9 NEBL come in instead, with .1 going to Treasury. And that would be even less if the node you delegated to has a fee % of their own. 

So for example, if 500,000 NEBL is the cold-staked TOTAL across the whole network, then 10% per year of 500k = 50k in rewards. Then 50k X the Treasury Tax of 10% = 5k NEBL

So in one year the Treasury would earn 5k NEBL.  This is not a ton of money but it could afford some things to be done at current prices.. But this can be significant IF the price of NEBL goes up dramatically.  What if 1 NEBL became worth $100?  That 5k per year is now equal to $500k USD to afford something significant. 

And the Treasury would just be a basic one.. nothing complicated.. just something the Neblio Team can tap into now and then to afford something extra needing to be paid. I don't feel we need to bother with any complicated governance over this little bit to secure it at this point, since we are early. We can always adjust this in the future if we want a more robust security on the Treasury & when NEBL's start to become worth something significant.. Then we can get more serious about it. so for now it would be fine to let the Neblio team handle the private key for this Treasury as it's just a paltry sum to afford some basic stuff or to tip themselves for some work they've done.  They could do NIP's to let us know they want to tap into it if they want for more clarity/communication but i don't find it a big deal just yet. I say lets get the ball rolling now and mold it better later.  Get the foundation there. 

Because the Treasury would not be able to afford much early, most of what we need will have to be done through charity or VC funding because we DO NOT want to change supply, nor do we want to over-tax the network.  No freebies are to be given to any team, unlike the previous NIP22.  The original Neblio team never got any handouts and neither should this one, nor any future teams.  If we handed them anything, they could bail on us and then the next team would want a hand out.. and the next.. and next... 

And, as far as inflation goes, we help battle it a tiny bit with a burn tax, which won't do much initially.. but if Neblio becomes used a lot one day, it will add up.  It seems to do very well on Ethereum & people like it. 
It doesn't have to be exactly as Eth's burn mechanism, because i realize Neblio may have some differences in it's design. But to give you a rough idea of what it does it gives a base-fee each TX must pay when people move funds or do any kind of activity. It would not affect the staking, just the end-users doing things.  Basically, gas fees will go up a bit compared to how they are now.

This NIP23 is a much better proposal for what I would call "Neblio 2.0" than that junk proposal NIP22. 

Oh and Note how i did not include "Smart Contracts".  Why?  Because we do not need to NIP that as it's something the original Neblio team was working on, so it's already gotten our approval when any of us first bought into the ICO.  They told us that was their goal. So it's something the new team can just go ahead and finish, as you can see here.

![image](https://user-images.githubusercontent.com/95306637/192832718-5a8b5950-2da6-4598-b2ed-b950b353c9a4.png)

### Proposal Voting Details

none yet

### Other Information

none yet.

**Comment by IWantMoreDecred** [comment]
Note: i am not a member of the Neblio team. I am just an investor and Node since 2018.

**Comment by RasmonT** [comment]
It's interesting, but what if I'm node and just use cold staking? I love cold staking because I don't need to have my hot wallet online 24/7, instead I delegated it to my node server without sending my Neblios directly to that server wallet, which is good for security measures. I wouldn't like to go back to the hot wallet staking model again, as I'm using my own node on the server and before I was using it on my old laptop that was running 24/7 (before cold staking). I agree that nodes should set up their own % of fee, but in my case it would be 0%.

Maybe add an option to tax addresses that aren't yours? Like I will be able to whitelist my address (because I'm running node) and the other delegators would be taxed because they're not running one? I think  this would be a good option.

**Comment by IWantMoreDecred** [comment]
That's a good point. I had forgotten we can do this. i never bothered doing it but i am aware of it. 

Had not considered that;  the treasury tax might be a bit of a problem then in that situation.

However, and you have come up with a good solution for the node fee system.. by simply white-listing addresses.
If we did that then the Neblio wallet should have 2 % fees then in the wallet settings somewhere.. The White-List Fee and the "Everyone Else Fee" field. 

As far as the treasury goes.. i'm thinking we should kick that can down the street then. I don't want to stifle nodes through taxation. 
The new team seems to be getting far more investment capital than any treasury would have provided in the confines of what we would be willing to do. So it would have only been a very modest treasury. pay for little things generally.

**Comment by IWantMoreDecred** [comment]
oh wait a second.. it is possible to do the treasury tax in the same way as the fee. 

i forgot that a Treasury tax like this would be processed by the same node who is collecting the reward, whether proxy or not, for the NEBL staked to their node.. So just as your node is a Fee collector, it is also the Tax Collector to collect on behalf of the treasury and then auto-send to the Treasury as part of the processing of the block. 
The 10% Treasury tax would simply follow your White-List exception and would assume that if you White-Listed an address for Feeless Cold-Staking, that you would also own that address and would not want it subject to the TTax either. 

So the white-list solves everything.

The only problem i could see is that some nodes may White-List people for TreasuryTax Evasion, but then again, that would mean the owner of the node is also not collecting any fees.. because white-listing would apply to both things. 

Maybe for a friend you don't want to charge fees to also?

But, for the most part, nodes will want to collect fees of some kind, so for those, they clearly wouldn't be whitelisted.. which means the Treasury Tax would also hit that. all processed by that node. 

Your Server Node would be directing the fees you earned from others to your real wallet.. the cold-staked one.. and also to the treasury address the 10%.
